### PR TITLE
[PLAYER-18] api manager

### PR DESCRIPTION
**Pull request by jparez**
## Description

Add an `APIManager` to expose the player API.

This is a breaking change since `setupRendere`r now returns the API, not the `DeviceRendererFactory` object.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I've read & comply with the [contributing guidelines](https://github.com/Genymobile/genymotion-device-web-player/blob/main/CONTRIBUTING.md)
- [ ] I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] I have made corresponding changes to the documentation (README.md).
- [ ] I've checked my modifications for any breaking changes.

### CODE REVIEW

**Review by pgivel — CHANGES_REQUESTED** [review]
code looks great, some small suggestions on the README

**Review by pgivel — APPROVED** [review]

File: src/APIManager.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    listRegisteredFunctions() {
+        const exposedFunctionsDescription = Object.entries(this.apiFunctions).reduce((acc, val) => {
+            acc[val[0]] = val[1].description;
+            return acc;
+        }, {});
+        return exposedFunctionsDescription;
+    }
+
+    getExposedApiFunctions() {
```
Comment (caarmen) [inline src/APIManager.js:65]: Can we have JSDoc indicating the types of objects which are returned, in the functions which return something?

File: src/APIManager.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            (json) => {
+                this.instance.sendEvent(json);
+            },
+            'send data to WS messages of player, must be a JSON object. Example: {type: "MOUSE_PRESS", x: 100, y: 100}',
+        );
+
+        // record fn to get registered functions
+        this.registerFunction(
+            'getRegisteredFunctions',
+            () => this.listRegisteredFunctions(),
```
Comment (caarmen) [inline src/APIManager.js:20]: Could we have the same name in both places? Either `getRegisteredFunctions` or `listRegisteredFunctions`, but not both at the same time?

File: README.md [hunk 3]
```diff
@@ -103,19 +103,49 @@ or check the [PaaS documentation](https://docs.genymotion.com/paas/01_Requiremen
...
     window.addEventListener('beforeunload', function () {
-        renderer.disconnect();
+        playerAPI.disconnect();
     });
 </script>
 ```
 
+## Player API
+
+plugin options and websocket communication can be handle trought the API object returned by setupRenderer function.
```
Comment (caarmen) [inline README.md:121]:
```suggestion
Plugin options and websocket communication can be handled through the API object returned by the `setupRenderer` function.
```

File: README.md [hunk 4]
```diff
@@ -103,19 +103,49 @@ or check the [PaaS documentation](https://docs.genymotion.com/paas/01_Requiremen
...
+
+Built-in exposed functions are
+
+### `getRegisteredFunctions`
+
+which return the list of available functions with an optionnal description
```
Comment (caarmen) [inline README.md:127]:
```suggestion
which returns the list of available functions with an optional description
```

File: README.md [hunk 5]
```diff
@@ -103,19 +103,49 @@ or check the [PaaS documentation](https://docs.genymotion.com/paas/01_Requiremen
...
+
+### `disconnect`
+
+which disconnect player from VM and cleanup memory listener
```
Comment (caarmen) [inline README.md:131]:
```suggestion
which disconnects the player from the VM and cleanups the memory listener
```

File: README.md [hunk 6]
```diff
@@ -103,19 +103,49 @@ or check the [PaaS documentation](https://docs.genymotion.com/paas/01_Requiremen
...
+
+### `addEventListener`
+
+used to listen messages emit from VM such as 'fingerprint', 'gps', 'BATTERY_LEVEL'
```
Comment (caarmen) [inline README.md:135]:
```suggestion
used to listen to messages emitted from the VM such as 'fingerprint', 'gps', 'BATTERY_LEVEL'
```

File: README.md [hunk 7]
```diff
@@ -103,19 +103,49 @@ or check the [PaaS documentation](https://docs.genymotion.com/paas/01_Requiremen
...
+
+```html
+addEventListener('fingerprint', (msg)=>{ console.log(msg) })
+```
+
+### `sendData`
+
+used to send message to VM.
```
Comment (caarmen) [inline README.md:143]:
```suggestion
used to send messages to the VM.
```

**Review by caarmen — APPROVED** [review]
I don't have the full context to review beyond a surface level.

Just checked the documentation mostly, which LGTM :)

**Pull request merged**